glue/qt/widgets/histogram_widget.py

```diff
@@ -193,6 +183,21 @@ def component(self, component):
 
     @defer_draw
     def _set_attribute_from_combo(self, *args):
+        if self.component is not None:
+            for d in self._data:
+                try:
+                    component = d.get_component(self.component)
+                except:
+                    continue
+                else:
+                    break
```

Comment:
Just to be clear, if a componentID belongs to multiple datasets, the actual component is the same though, right? So choosing any of them should be ok?